Assets/Scripts/MonoBehaviours/EnemyController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using MonoBehaviours.WeaponsSystem;
+using ScriptableObjects.EnemiesSystem;
+using ScriptableObjects.EnemiesSystem.EnemyAttackStrategies;
+using ScriptableObjects.EnemiesSystem.EnemyMovementStrategies;
+using UnityEngine;
+
+namespace MonoBehaviours
+{
+    public class EnemyController : MonoBehaviour
+    {
+        #region Interface Variables
+
+        [SerializeField] private EnemyBehaviourData enemyBehaviourData;
+        [SerializeField] private float jumpForce = 300;
+
+        #endregion
+
+        #region Private Variables
+
+        private GameObject _player;
+        private Vector3 _lastPosition;
+        private float _timer = 2;
+        private bool _hasCollided;
+        private EnemyAttackStrategy _attackStrategy;
+        private EnemyMovementStrategy _movementStrategy;
+
+        #endregion
+
+        void Start()
+        {
+            _attackStrategy = enemyBehaviourData.enemyAttackStrategy;
+            _movementStrategy = enemyBehaviourData.enemyMovementStrategy;
+            _player = GameObject.FindWithTag("Player");
+        }
+
+        void Update()
+        {
+            _movementStrategy.Move(gameObject.transform, _player.transform);
+            _timer -= Time.deltaTime;
+            CheckIfStuck();
+            IsFacingPlayer();
+            _hasCollided = _attackStrategy.Attack(_player, gameObject, enemyBehaviourData.damageGiven, _hasCollided);
+            _lastPosition = gameObject.transform.position;
+        }
+
+        private void IsFacingPlayer()
+        {
+            var emuPosition = gameObject.transform.position;
+            var playerPosition = _player.transform.position;
+
+            gameObject.GetComponent<SpriteRenderer>().flipX = (emuPosition.x - playerPosition.x < 0);
+            if (gameObject.GetComponentInChildren<WeaponBehaviourScript>() != null)
+            {
+                gameObject.GetComponentInChildren<WeaponBehaviourScript>().Direction =
+                    gameObject.GetComponent<SpriteRenderer>().flipX ? Vector2.right : Vector2.left;
+            }
+        }
+
+        private void OnTriggerEnter2D(Collider2D col)
+        {
+            if (col.gameObject.CompareTag("Projectile"))
+            {
+                Destroy(col.gameObject);
+                Destroy(gameObject);
+            }
+            else if (col.gameObject.CompareTag("Player"))
+            {
+                _hasCollided = true;
+            }
```

Comment:
I remember in the meeting you said that enemies have HP, so don't forget to handle collision with projectiles properly

Reply:
As of right now, we don't take the HP into account. I was going to do that in a second PR since it was becoming a lot of material in the PR, what do you think?

Reply:
Dealt with in issue #52